### ISSUE TITLE
Emoji Misplacement in Between Text

### ISSUE THREAD

**Issue by wei** [vercel/satori#305]
# Bug report

## Description / Observed Behavior

When Embed Font is turned on, emoji is misplaced when it's in between text.
When Embed Font is turned off, the font is unexpectedly changed.

## Expected Behavior

When Embed Font is turned on, emoji should not be misplaced.
When Embed Font is turned off, font should not be changed.

## Reproduction

[Example showing the issue with "Embed Font: True" 
<img width="1792" alt="image" src="https://user-images.githubusercontent.com/5880908/201497823-5eb39eb0-760c-46ee-af82-3e46adfbe6e0.png">
](https://og-playground.vercel.app/?share=dVI9T9xAEP0r440SGivnRIhEzocUXbhwRdIkpZs975y9l_WuZe9xZ05XGIGERIUQEh1QIAoKREFJRW9-hKGCgp_A7h0gPkQzu_Pe7OzsezsioWJIfNJoQIsK0aHhf-gqqXOgksF8onocaIbACkkTHpqSAoSiDBl0M5XAL6UigdB6ONL8-ScHLkHHPAeGiXofyEB-ZXwxkAC5LgR-G43sHoDxPBW08GGmK3A4407RGHkUawN-8Ly39-CAMx0_w1LKGJfRBE2H8NGEe4oKHsm2xiQ3bIhSY_aEapp5DfqC7PVzzbvFa7RVpmV0EHbmtmVcCMhvajacCmiHSuYBeVz9ly-hD7PeHaRxqH_YAV60tspHmepL1lRCZYYfxFzjlB6Pbfxuw83OxvrNzvYKLKBdhFAOXJzuXu6f2tSB6621q9UtB-pys17erMvDutyryyMHqoPquDo6L6uT85XqxIEF4-XZvgPv3gzx09zcFzAuNYxN5hbiEpVqbh5D_BGZSE_8z57nkqk3xJ-1CcNOPyK-zvroEuN0j_8rUvuZ9GCSmTZWgvmkg2xaNh7fAg)

[Example showing font weirdness with "Embed Font: False" 
<img width="1792" alt="image" src="https://user-images.githubusercontent.com/5880908/201509561-74014fc4-6d7b-493f-bd08-43901f7a36c0.png">
 ](https://og-playground.vercel.app/?share=dVI9b9swEP0rJxZtFqFWiyAt1A-gcOPGQ7u0oxZaPEl0KVIQ6dqq4UFFChToVAQBuqUdgg4dggwZM2VXfoSSKRnyE0LaSZAPZDny3jsej-9xSmLFkISk04EeFWJA48-QKGk0UMlgNVdDDrREYJWkOY9tSQVCUYYMklLl8E6pVCD0ro50337QwCWYjGtgmKvHkYzkS8a_RBJAm0rgq-nU7QEY14WgVQhLicDJkr9AM-RpZiz4JAgeXoJjzkx2CysoY1ymc7SYwFMbLikqeCr7BnNt2RilwfIG1bXzWvQOORxpw5PqPtop07M6CDdz3zE-ROQ9tRtOBfRjJXVErld_5F8xhOXgAjI4MW_cAHdaO-XTUo0k6yqhSsuPM25wQc9mLr524Wzr18-zrd_rsIZuEUJ5cLT_53h736UenG7-OPm-6UFbb7TfNtr6f1v_besdD5p_zW6zc1g3e4frzZ4Ha9bLg20PHj2Y4LOVlRdgXepYm-wtxCeqMNw-hoRTMpeehM-DwCcLb0i47BKGg1FKQlOO0CfW6SH_VBXuM5nxPLNtnASr-QAZCRMqNM5m5w)

## Additional Context

Satori version, and any other context about the problem here.

`satori@0.44.0`

**Comment by wei** [comment]
Duplicate of #302

**Comment by shuding** [comment]
The misplacement bug was fixed together with #302, but the font change is expected. In the playground, Satori has only Inter and Material Icons provided via `fonts`. And for the rest characters that can't be rendered, the playground uses the `loadAdditionalAsset` API to load the font from Google Fonts (Noto Fonts), and embed the rendered vector as SVG `path`.

If turning off `embedFont`, it will be a SVG `<text>` element and all characters will be rendered by the system using the `Inter, "Material Icons"` font family configuration. It's up to the system to fallback to a font to render the rest characters such as Kanji.

**Comment by ducan-ne** [comment]
@shuding I'm not sure if my issue is related to this. Below is what I got when running the example in the playground with `embedFont: false`, It's misplacement. If it's unrelated, I'll to create a new issue.
```html
<svg width="400" height="200" viewBox="0 0 400 200" xmlns="http://www.w3.org/2000/svg"><mask id="satori_om-id"><rect x="0" y="0" width="400" height="200" fill="#fff"/></mask><rect x="0" y="0" width="400" height="200" fill="#fff"/><clipPath id="satori_cp-id-0"><rect x="163" y="28" width="75" height="65"/></clipPath><mask id="satori_om-id-0"><rect x="163" y="28" width="75" height="65" fill="#fff"/></mask><image x="163" y="28" width="75" height="65" href="data:image/svg+xml;utf8,%3Csvg  fill=%22%23000%22 xmlns=%22http://www.w3.org/2000/svg%22 width=%22NaN%22 height=%22null%22 viewBox=%220 0 75 65%22%3E%3Cpath d=%22M37.59.25l36.95 64H.64l36.95-64z%22%3E%3C/path%3E%3C/svg%3E" preserveAspectRatio="none" clip-path="url(#satori_cp-id-0)" mask="url(#satori_om-id-0)"/><mask id="satori_om-id-0-0"><rect x="163" y="28" width="0" height="65" fill="#fff"/></mask><mask id="satori_om-id-1"><rect x="104" y="133" width="193" height="39" fill="#fff"/></mask><text x="104" y="162.9522109310629" width="80.04545454545455" height="38.72727272727273" font-weight="600" font-style="normal" font-size="32" font-family="serif" fill="black">Hello</text><text x="184.04545454545456" y="162.9522109310629" width="9.693181818181818" height="38.72727272727273" font-weight="600" font-style="normal" font-size="32" font-family="serif" fill="black">,</text><text x="193.73863636363637" y="162.9522109310629" width="7.420454545454546" height="38.72727272727273" font-weight="600" font-style="normal" font-size="32" font-family="serif" fill="black"> </text><text x="201.15909090909093" y="162.9522109310629" width="93.01136363636364" height="38.72727272727273" font-weight="600" font-style="normal" font-size="32" font-family="serif" fill="black">World</text></svg>
```

```tsx
console.log(
  await satori(
    <div
      style={{
        height: '100%',
        width: '100%',
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: '#fff',
        fontSize: 32,
        fontWeight: 600,
      }}
    >
      <svg
        width="75"
        viewBox="0 0 75 65"
        fill="#000"
        style={{ margin: '0 75px' }}
      >
        <path d="M37.59.25l36.95 64H.64l36.95-64z"></path>
      </svg>
      <div style={{ marginTop: 40 }}>Hello, World</div>
    </div>,
    {
      width: 400,
      height: 200,
      fonts: [{
        name: 'Interx',
        data: font,
        weight: 400,
        style: 'normal',
      }],
      embedFont: false,
    },
  ),
  ```
